magento/devdocs#: Layout instructions. Adding missing Metadata interfaces info. [GITBUG]

DenisSaltanahmedov opened the pull request:
## Purpose of this pull request
This pull request (PR) Adding missing information about Metadata interfaces.
The Service contract design patterns topic provide a list of types of service interfaces
-  Repository interfaces
-  Management interfaces
-  Metadata interfaces
Now the "Metadata interfaces" item is not described in the topic.

## Affected DevDocs pages
-  https://devdocs.magento.com/guides/v2.3/extension-dev-guide/service-contracts/design-patterns.html

## Links to Magento source code
-  https://github.com/magento/devdocs/blob/master/src/guides/v2.3/extension-dev-guide/service-contracts/design-patterns.md

whatsnew
Added the 'Metadata interfaces' section to the [Service contract design patterns](https://devdocs.magento.com/guides/v2.3/extension-dev-guide/service-contracts/design-patterns.html#metadata-interfaces) topic.

devops-devdocs commented:
An admin must run tests on this PR before it can be merged.

rogyar commented on src/guides/v2.3/extension-dev-guide/service-contracts/design-patterns.md:
```diff
@@ -148,6 +148,49 @@ Management interfaces provide management functions that are not related to repos
+
+Metadata interfaces provide methods for retive metadata, the interfaces are not related to repositories. For example:
+
+<table style="width:100%">
```
Hi @DenisSaltanahmedov. Please, feel free to use Markdown tables instead of HTML tables. It's a preferable way.

DenisSaltanahmedov commented:
Hi @rogyar, thanks for your review.
I have fixed the recommendations. Also, I changed Markdown tables instead of HTML for all tables in this topic.
Thanks.

rogyar commented on src/guides/v2.3/extension-dev-guide/service-contracts/design-patterns.md:
```diff
@@ -69,85 +69,35 @@ For example, persistent data entities for the Customer module include Customer,
-      </td>
-   </tr>
-</table>
+Interface | Description
```
```suggestion
Method | Description
```

rogyar commented on src/guides/v2.3/extension-dev-guide/service-contracts/design-patterns.md:
```diff
@@ -69,85 +69,35 @@ For example, persistent data entities for the Customer module include Customer,
+--- | ---
+`save(data entity interface)` | If an ID is not specified, creates a record, If an ID is specified, updates the record for the specified ID.
+`get(id)` | Performs a database lookup by ID and returns a data entity interface, such as `CustomerInterface`or `AddressInterface`.
+`getList(search criteria)` | Performs a search for all data entities that match specified search criteria and returns a search results interface that gives access to the set of matches.
```
```suggestion
`getList(search criteria)` | Performs a search for all data entities that match specified search criteria and returns a search result interface that gives access to the set of matches.
```

rogyar commented on src/guides/v2.3/extension-dev-guide/service-contracts/design-patterns.md:
```diff
@@ -69,85 +69,35 @@ For example, persistent data entities for the Customer module include Customer,
 
+### Metadata interfaces {#metadata-interfaces}
+
+Metadata interfaces provide methods for retive metadata, the interfaces are not related to repositories. For example:
```
```suggestion
Metadata interfaces provide methods for retrieving metadata, the interfaces are not related to repositories. For example:
```

rogyar commented on src/guides/v2.3/extension-dev-guide/service-contracts/design-patterns.md:
```diff
@@ -69,85 +69,35 @@ For example, persistent data entities for the Customer module include Customer,
+`AttributeMetadataInterface` | Provides customer attribute metadata and defines the constants used as keys of data array and methods. See more [AttributeMetadataInterface]({{ site.mage2bloburl }}/{{ page.guide_version }}/app/code/Magento/Customer/Api/Data/AttributeMetadataInterface.php){:target="_blank"}.
+`ProductMetadataInterface` | Provides Magento application product metadata. Defines the `getVersion()`, `getEdition()`, `getName()` methods.
+`CustomerMetadataManagementInterface` | Interface for managing customer attributes metadata. Defines the constant `ENTITY_TYPE_CUSTOMER`
+`AddressMetadataInterface` | Interface for retrieval information about customer address attributes metadata. Defines the constants `ATTRIBUTE_SET_ID_ADDRESS`, `ENTITY_TYPE_ADDRESS`, `DATA_INTERFACE_NAME`
```
```suggestion
`AddressMetadataInterface` | Interface for retrieving information about customer address attributes metadata. Defines the constants `ATTRIBUTE_SET_ID_ADDRESS`, `ENTITY_TYPE_ADDRESS`, `DATA_INTERFACE_NAME`
```

DenisSaltanahmedov commented:
Hi @rogyar, fixed.
Thanks.

dobooth commented:
Love getting rid of HTML. Thanks!

dobooth commented:
Just a note that we have a strange bug in our repo that is causing some PRs to not update. We will merge as soon as we get it solved.

DenisSaltanahmedov commented:
Hi @dobooth, any updates about the issue? Thanks.

jeff-matthews commented:
@DenisSaltanahmedov, the **Update branch** button isn't working. We've opened a ticket with GitHub support to fix it. If we can't get support to help us by next week, we'll start using workarounds to merge all affected PRs.

DenisSaltanahmedov commented:
Hi, if we still have the problem with merge in the PR, I can recreate the changes to new PR from the current master branch. Thanks.

dshevtsov commented:
@DenisSaltanahmedov Looks like the issue was fixed finally. Thank you for your patience.

dobooth commented:
running tests
